packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx

```diff
@@ -45,8 +47,21 @@ export function TreeRenderer({
       return (
         <TreeNodeRenderer
           {...nodeProps}
+          onNodeClick={(nodeId, isSelected, event) => {
+            // Ignore double clicks
+            onNodeDoubleClick && event.detail !== 2 && onNodeClick?.(nodeId, isSelected, event);
+          }}
+          nodeProps={{
+            onDoubleClick: (event) => {
+              if (nodeProps.isDisabled || "type" in nodeProps.node || !isPresentationHierarchyNode(nodeProps.node)) {
+                return;
+              }
+              onNodeDoubleClick?.(nodeProps.node, !!nodeProps.isSelected);
+              // Select node to not lose highlight
+              !nodeProps.isSelected && onNodeClick?.(nodeProps.node, true, event);
```

Comment:
does it only loose selection in the tree or does this cause the node to be deselected and afterwards selected in unified selection?

Reply:
The first click is handled as normal so it does deselect in unified selection if double clicked slowly enough. Seems to work the same as in the old models tree.

Reply:
As discussed, in "extended" selection mode a click on a selected node should not deselect it. I think that solves the problem and we won't need to select it back.

In other selection modes let's keep the behavior you have now.

Reply:
Done.

Reply:
For tracking - the change was made in https://github.com/iTwin/presentation/pull/627 and consumed into tree widget with https://github.com/iTwin/viewer-components-react/pull/898.